src/picasso-definition/components/navigation-panel/index.js

```diff
@@ -0,0 +1,101 @@
+import NUMBERS from '../../../constants/numbers';
+import KEYS from '../../../constants/keys';
+
+// Translation (up/down/left/right)
+function translate({ viewHandler, direction, percent }) {
+  const { xAxisMax, xAxisMin, yAxisMax, yAxisMin } = viewHandler.getDataView();
+  let deltaX = 0;
+  let deltaY = 0;
+  if (direction === 'x') {
+    deltaX = ((xAxisMax - xAxisMin) * percent) / 100;
+  } else {
+    deltaY = ((yAxisMax - yAxisMin) * percent) / 100;
+  }
+  const dataView = {
+    xAxisMin: xAxisMin + deltaX,
+    xAxisMax: xAxisMax + deltaX,
+    yAxisMax: yAxisMax + deltaY,
+    yAxisMin: yAxisMin + deltaY,
+  };
+  viewHandler.setDataView(dataView);
+}
+
+// Zoom in/ Zoom out
+function transform(scale, start, end, factor) {
+  const range = end - start;
+  const value = start + range * scale;
+  const newStart = value - range * scale * factor;
+  const newEnd = newStart + range * factor;
+  return [newStart, newEnd];
+}
+
+function zoom({ viewHandler, zoomDirection }) {
+  const { scale, maxScale, minScale } = viewHandler.getMeta();
+  const zoomFactor = zoomDirection === 'in' ? 2 ** (-1 / 2) : 2 ** (1 / 2);
+  const newScale = zoomFactor * scale;
+
+  if (newScale > maxScale || newScale < minScale) {
+    return;
+  }
+  const { xAxisMin, xAxisMax, yAxisMin, yAxisMax } = viewHandler.getDataView();
+  const [xMin, xMax] = transform(0.5, xAxisMin, xAxisMax, zoomFactor);
+  const [yMax, yMin] = transform(0.5, yAxisMax, yAxisMin, zoomFactor);
+  viewHandler.setDataView({
+    xAxisMin: xMin,
+    xAxisMax: xMax,
+    yAxisMin: yMin,
+    yAxisMax: yMax,
+  });
+  viewHandler.setMeta({ scale: newScale });
+}
+
+export default function createNavigationPanel({ layoutService, chartModel }) {
+  const navigation = layoutService.getLayoutValue('navigation');
+  const viewHandler = chartModel.query.getViewHandler();
+  return {
+    key: KEYS.COMPONENT.NAVIGATION_PANEL,
+    type: 'navigation-panel',
+    show: navigation,
+    style: { padding: NUMBERS.NAVIGATION_PANEL.PADDING, gridWidth: NUMBERS.NAVIGATION_PANEL.GRID_WIDTH },
+    buttonList: {
```

Comment:
We rarely suffix options/variables like this, `buttons` should be enough.  But would probably rename it to `actions`.

The API below feels a bit bloated, that the nested `callBacks` are needed (should be `callback` with lower case `b` by the way 😄 ), but will have to think about a better solution for a while (that home needs `isDisabled` makes it harder to reorganize).

Reply:
A cleaner way that conforms more with picasso and other things we have would be:

```js
on: {
  home: () => {},
  zoomIn: () => {},
  zoomOut: () => {},
}
```

Reply:
`buttonList` and some other things should also be placed under a `settings` object. Take a look at how other picasso components do. It's good if we try to be as consistent as we can 👍

Reply:
how about this?
```{
    key: KEYS.COMPONENT.NAVIGATION_PANEL,
    type: 'navigation-panel',
    show: navigation,
    style: { padding: NUMBERS.NAVIGATION_PANEL.PADDING, gridWidth: NUMBERS.NAVIGATION_PANEL.GRID_WIDTH },
    settings: {
      actions: {
        home: () => {
          viewHandler.setDataView(viewHandler.getMeta().homeStateDataView);
        },
        left: () => {
          translate({ viewHandler, direction: 'x', percent: -10 });
        },
        right: () => {
          translate({ viewHandler, direction: 'x', percent: 10 });
        },
        up: () => {
          translate({ viewHandler, direction: 'y', percent: 10 });
        },
        down: () => {
          translate({ viewHandler, direction: 'y', percent: -10 });
        },
        zoomIn: () => {
          zoom({ viewHandler, zoomDirection: 'in' });
        },
        zoomOut: () => {
          zoom({ viewHandler, zoomDirection: 'out' });
        },
      },
      isDisabled: {
        home: () => viewHandler.getMeta().isHomeState,
      },
    },
  }```